src/pkg/packager/deploy.go

```diff
@@ -162,10 +163,16 @@ func (p *Packager) deployComponents(ctx context.Context) ([]types.DeployedCompon
 			if err := p.connectToCluster(connectCtx); err != nil {
 				return nil, fmt.Errorf("unable to connect to the Kubernetes cluster: %w", err)
 			}
+			// If this package has been deployed before, increment the package generation within the secret
+			if existingDeployedPackage, _ := p.cluster.GetDeployedPackage(ctx, p.cfg.Pkg.Metadata.Name); existingDeployedPackage != nil {
+				packageGeneration = existingDeployedPackage.Generation + 1
+			}
 		}
 
 		deployedComponent := types.DeployedComponent{
-			Name: component.Name,
+			Name:               component.Name,
+			Status:             types.ComponentStatusDeploying,
```

Comment:
component status deploying is not used. Note that in the original implementation the package secret was updated before the deployment started. 

This also means you will have to update the test statements in the actions at `src/test/packages/22-git-data/zarf.yaml`

Reply:
I'm not sure what you mean by not used. This updates the component to "deploying" before the deploy actually starts then will transition to failed or succeeded based on the actual deployComponent call on line 197. 

This shouldn't change the length of the deployedComponents list, just adds back the Status field so I think that the test statements in `src/test/packages/22-git-data/zarf.yaml` are still correct

Reply:
I can add an update to that test to cover the deployedComponent status though and make sure it's correct!

Reply:
Being able to "catch" a "deploying" status is going to be tough for the e2e tests

Reply:
Yeah but updating the `deployedComponents` object doesn't update the secret in the cluster. We'll need a call to `p.cluster.RecordPackageDeployment(ctx, p.cfg.Pkg, deployedComponents)` before the deployment starts to do that. 

Once we start making the call to `p.cluster.RecordPackageDeployment(ctx, p.cfg.Pkg, deployedComponents)` before the component is deployed, the numbers in the test statements of `src/test/packages/22-git-data/zarf.yaml` will increase by 1 because we'll have started tracking the deployment before the actions start rather than after

Reply:
Ah ok makes sense!